src/datasets/features/audio.py

```diff
@@ -92,7 +94,23 @@ def encode_example(self, value: Union[str, dict]) -> dict:
             return {"bytes": buffer.getvalue(), "path": None}
         elif value.get("path") is not None and os.path.isfile(value["path"]):
             # we set "bytes": None to not duplicate the data if they're already available locally
-            return {"bytes": None, "path": value.get("path")}
+            if value.get("path").endswith("pcm"):
```

Comment:
you can access "path" directly, since we checked above that it exists
```suggestion
            if value["path"].endswith("pcm"):
```